cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,132 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
     }
 }
 
+void CallProfilerSerializer::serialize_call_profiler(const CallProfiler & profiler, std::ostream & outstream)
+{
+    // Example:
+    //  {
+    //      "radix_tree":
+    //      {
+    //          "nodes":[
+    //              {"key":-1,"name":"","data":{"start_time": 0,"caller_name": "","total_time": 0,"call_count": 0,"is_running": 0},"children":[0]},
+    //              {"key":0,"name":"void modmesh::detail::foo1()","data":{"start_time": 17745276708555250,"caller_name": "void modmesh::detail::foo1()","total_time": 61002916,"call_count": 1,"is_running": 1},"children":[1]},
+    //              {"key":1,"name":"void modmesh::detail::foo2()","data":{"start_time": 17745276708555458,"caller_name": "void modmesh::detail::foo2()","total_time": 54002250,"call_count": 1,"is_running": 1},"children":[2]},
+    //              {"key":2,"name":"void modmesh::detail::foo3()","data":{"start_time": 17745276743555833,"caller_name": "void modmesh::detail::foo3()","total_time": 19001833,"call_count": 1,"is_running": 1},"children":[]}
+    //          ],
+    //          "current_node":-1,
+    //          "unique_id":3
+    //      }
+    //  }
+
+    outstream << R"({)";
+    outstream << R"("radix_tree": )";
+    CallProfilerSerializer::serialize_radix_tree(profiler, outstream);
+    outstream << R"(})";
+}
+
+void CallProfilerSerializer::serialize_radix_tree(const CallProfiler & profiler, std::ostream & outstream)
+{
+    // Example:
+    //  {
+    //      "nodes":[
+    //          {"key":-1,"name":"","data":{"start_time": 0,"caller_name": "","total_time": 0,"call_count": 0,"is_running": 0},"children":[0]},
+    //          {"key":0,"name":"void modmesh::detail::foo1()","data":{"start_time": 17745276708555250,"caller_name": "void modmesh::detail::foo1()","total_time": 61002916,"call_count": 1,"is_running": 1},"children":[1]},
+    //          {"key":1,"name":"void modmesh::detail::foo2()","data":{"start_time": 17745276708555458,"caller_name": "void modmesh::detail::foo2()","total_time": 54002250,"call_count": 1,"is_running": 1},"children":[2]},
+    //          {"key":2,"name":"void modmesh::detail::foo3()","data":{"start_time": 17745276743555833,"caller_name": "void modmesh::detail::foo3()","total_time": 19001833,"call_count": 1,"is_running": 1},"children":[]}
+    //      ],
+    //      "current_node":-1,
+    //      "unique_id":3
+    //  }
+
+    outstream << R"({)";
+    outstream << R"("nodes": [)";
+    CallProfilerSerializer::serialize_radix_tree_nodes(profiler.radix_tree().get_current_node(), outstream);
+    outstream << R"(],)";
+    outstream << R"("current_node": )" << profiler.radix_tree().get_current_node()->key() << R"(,)";
+    outstream << R"("unique_id": )" << profiler.radix_tree().get_unique_node();
+    outstream << R"(})";
+}
+void CallProfilerSerializer::serialize_radix_tree_nodes(const RadixTreeNode<CallerProfile> * node, std::ostream & outstream)
```

Comment:
Keep a blank line between functions.

Is it possible to make clang-format capture this?  (Apparently we haven't configured it this way, yet.)